data/transactions/payment.go

```diff
@@ -37,6 +37,22 @@ type PaymentTxnFields struct {
 	CloseRemainderTo basics.Address `codec:"close"`
 }
 
+// payment captures fields for one payment of a multi-payment transaction.
+type payment struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	PaymentTxnFields `codec:"mxpay"`
+	Sender           basics.Address `codec:"mxsnd"`
```

Comment:
You might consider having multipayment transactions have multiple recipients but a single sender -- this way it's clear which address to check the signature against and which address the fee comes from. Depending on use case, using a multi-recipient transaction to spend out of an n-of-n multisig address might be equivalent to a multi-sender multi-receiver transaction.

Reply:
The fee is paid by the Header.Sender that initiates the transaction, so that's not a problem.  It would certainly be much easier to have one sender to check signatures against, but the spec from the start was to handle arbitrary lists of sender-receiver pairs, and to support but not require multisig accounts.